### PR TITLE
PMM-7231 IA Disable channel

### PR DESCRIPTION
**Pull request by Dasio**
[PMM-7231](https://jira.percona.com/browse/PMM-7231)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/1537

- [ ] Links to other linked pull requests (optional).
---
- [ ] Tests passed.
- [ ] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

File: services/alertmanager/alertmanager.go [hunk 1]
```diff
@@ -359,20 +359,33 @@ func (svc *Service) populateConfig(cfg *alertmanager.Config) error {
 		cfg.Global = &alertmanager.GlobalConfig{}
 	}
 
-	// make sure that "empty" receiver is there
-	var emptyFound bool
-	for _, r := range cfg.Receivers {
-		if r.Name == "empty" {
-			emptyFound = true
-			break
+	findReceiver := func(name string) int {
```
Comment (BupycHuk) [inline services/alertmanager/alertmanager.go:362]: findReceiverIdx?

File: services/alertmanager/alertmanager.go [hunk 2]
```diff
@@ -359,20 +359,33 @@ func (svc *Service) populateConfig(cfg *alertmanager.Config) error {
...
 			Name: "empty",
 		})
 	}
 
+	disabledReceiver := &alertmanager.Receiver{
+		Name: "disabled",
+	}
+	disabledReceiverIndex := findReceiver("disabled")
+	if disabledReceiverIndex >= 0 {
+		cfg.Receivers[disabledReceiverIndex] = disabledReceiver
```
Comment (BupycHuk) [inline services/alertmanager/alertmanager.go:384]: why do we update it here?

File: services/alertmanager/alertmanager.go (hunk 1 above)
Comment (BupycHuk) [reply services/alertmanager/alertmanager.go:362]: based on the question below looks like we can return boolean here

File: services/alertmanager/alertmanager.go (hunk 2 above)
Comment (Dasio) [reply services/alertmanager/alertmanager.go:384]: so user can't add his "disabled" receiver by using file.
Same thing as he can add "empty" (default) receive.

Comment (BupycHuk) [reply services/alertmanager/alertmanager.go:384]: Yeah, but why do we replace it?

Comment (Dasio) [reply services/alertmanager/alertmanager.go:384]: Any better solution how to forbid user to add this receiver?
Check after yaml is parsed? And what then?

Comment (BupycHuk) [reply services/alertmanager/alertmanager.go:384]: ok, so we let users to override empty, but don't let to override disabled, right?
If it is, then everything is ok, but we should document it.

Comment (artemgavrilov) [reply services/alertmanager/alertmanager.go:384]: I agree, better to add a comment here.

**Review by artemgavrilov — APPROVED** [review]
Please add comments about `disabled` receivers/

Other stuff LGTM

**Review by oter — APPROVED** [review]

File: services/alertmanager/alertmanager.go [hunk 3]
```diff
@@ -359,20 +359,32 @@ func (svc *Service) populateConfig(cfg *alertmanager.Config) error {
...
+	if findReceiverIdx("empty") == -1 {
 		cfg.Receivers = append(cfg.Receivers, &alertmanager.Receiver{
 			Name: "empty",
 		})
 	}
 
+	disabledReceiver := &alertmanager.Receiver{
+		Name: "disabled",
+	}
+	// Override if there is any user defined receiver `disabled, needs to be empty
```
Comment (BupycHuk) [inline services/alertmanager/alertmanager.go:381]:
```suggestion
	// Override if there is any user defined receiver `disabled`, needs to be empty
```

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**